### PR TITLE
feat: add terminal titles with statuses

### PR DESCRIPTION
**Pull request by TylerBarnes**
Had a few mins over my lunch to work on this :)
It's always bugged me how when I'm running node scripts it just says "node" in the terminal title. This PR adds which command is running as well as a status indicator (running, success, failed, cancelled).

I've only tested this on macOS (I can try linux/windows as well later)

Examples in VSCode terminal:

running
Screenshot 2023-04-12 at 12 04 45 PM

cancelled
Screenshot 2023-04-12 at 12 05 30 PM

failed
Screenshot 2023-04-12 at 12 06 43 PM

success 
Screenshot 2023-04-12 at 12 05 57 PM

swpx too
Screenshot 2023-04-12 at 12 09 49 PM

The cool thing is atleast with VSCode when using terminal editor tabs (newer feature) if the terminal is the active tab it shows it in the app title bar as well as if you right click the VSCode icon in the macOS dock and see the open window list
 
Screenshot 2023-04-12 at 12 04 58 PM

I'm not really sure how to add a test for this 😅 maybe it's not possible?

### CODE REVIEW

File: src/helpers/cmds.js [hunk 1]
```diff
@@ -26,6 +29,47 @@ const addPositional = (args, action) => {
...
+    throw new Error(`Terminal title status must be provided.`)
+  }
+
+  titleState.status = status
+
+  if (!titleState.cliName || !titleState.packageManagerName || !titleState.action) {
+    throw new Error(`Terminal title must be initialized before updating it.`)
+  }
+
+  setTitle(`${titleState.status} ${titleState.cliName}: ${titleState.packageManagerName} ${titleState.action}`)
```
Comment (TylerBarnes) [inline src/helpers/cmds.js:57]: I'm used to using prettier but I don't see a prettierrc, how should I format?

**Comment by TylerBarnes** [comment]
I should add this is also not vscode-specific, it should generally work across different terminal emulators.

Iterm2:
Screenshot 2023-04-12 at 12 24 31 PM

Screenshot 2023-04-12 at 12 23 34 PM

macOS terminal:
Screenshot 2023-04-12 at 12 25 48 PM

Screenshot 2023-04-12 at 12 26 04 PM

File: src/helpers/cmds.js (hunk 1 above)
Comment (equiman) [reply src/helpers/cmds.js]: I'm only using eslint on this project with these configurations on `.vscode\settings.json` file.

```json
"[javascript][typescript]": {
    "editor.defaultFormatter": "dbaeumer.vscode-eslint",
    "editor.formatOnSave": false,
    "editor.codeActionsOnSave": [
      "source.formatDocument",
      "source.fixAll.eslint"
    ],
  },
```

I think you need to have installed the [ESLint](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint) extension to autoformat the code on save.

**Comment by equiman** [comment]
I didn't think about that. It's an excellent feature.
I'll try to find time to take a look tonight and release a new version. 

Thanks!

**Comment by equiman** [comment]
For me, it's an improvement, it will be included on the core.

I will try to finish something quickly and I'll take a look and test at these cool new features.

**Comment by TylerBarnes** [comment]
Awesome, thank you! btw just pushed up another commit because I realized the clock "animation" is a bit buggy for some terminals (the tab flashes when re-rendering the title)

**Comment by equiman** [comment]
I've tested on Ubuntu (WSL) with Windows Terminal and almost is showing the message with the watch icon:
image

But once finished shows the path again as title:
image

**Comment by equiman** [comment]
Please test again on macOS maybe I forget something while doing the refactoring 😅

I've moved some functions and renamed the parameters to follow the same names used on the other functions or the name that uses Yargs for that.

**Comment by equiman** [comment]
Please test again on macOS maybe I forget something while doing the refactoring 😅

I've moved some functions and renamed the parameters to follow the same names used on the other functions or the name that uses Yargs for that.

**Comment by equiman** [comment]
Seems to be that the `node-bash-title` package only includes the macOS version. 

Tomorrow I'll try with this solution:

https://stackoverflow.com/questions/29548477/how-do-you-set-the-terminal-tab-title-from-node-js/75469151#75469151

**Comment by equiman** [comment]
Well definitely seem to be that `npm` also is changing the title and I didn't find a way to disable this behavior.

Please test it again on macOS and if it works I'll release it as a new version.

**Comment by TylerBarnes** [comment]
Apologies for the delay on this! I've been really busy but I haven't abandoned this

**Comment by equiman** [comment]
Don't worry, when you can test it. Just let me know if works well on Mac OS.

**Comment by TylerBarnes** [comment]
@equiman I'm seeing what you're seeing now. Not sure how I was seeing it persist after it was finished before 🤔 Tried a bunch of node versions and a few package managers and saw the same behaviour across different terminals. So it only works while the process is running.
Do you think it's still worth having then or should I close this?

**Comment by equiman** [comment]
If it only works while the commands are running, I didn't see much value in that. 

But we can spend the last round. Can you test if it works again previous to my changes?
I can review again the code starting from that point if you want.

**Comment by TylerBarnes** [comment]
I agree with that! Too bad it doesn't work, I'm quite confused how it did before 😅 I went back and checked out my initial commits and it's not working there either. Very strange. To me that implies there is some way for it to work but I wasn't able to figure out how and don't have more time to figure it out unfortunately.
So I'll close this for now. I may re-open if I eventually get some time to investigate further and figure out how to make it work.

Thanks for all the help on this @equiman !

**Pull request closed without merging**